mica/archive/cda/services.py

```diff
@@ -0,0 +1,665 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Python interface to the Chandra Data Archive (CDA) web services and an
+interface to a local disk copy of the Observation Catalog (Ocat).
+"""
+from pathlib import Path
+import re
+import warnings
+import time
+
+import requests
+import numpy as np
+import tables
+from astropy.table import Table, MaskedColumn
+from astropy.coordinates import SkyCoord
+
+from mica.common import MICA_ARCHIVE
+
+
+__all__ = ['get_archive_file_list', 'get_proposal_abstract',
+           'get_ocat_web', 'get_ocat_local']
+
+OCAT_TABLE_PATH = Path(MICA_ARCHIVE) / 'ocat_target_table.h5'
+OCAT_TABLE_CACHE = {}
+
+URL_CDA_SERVICES = "https://cda.harvard.edu/srservices"
+CDA_SERVICES = {
+    'prop_abstract': 'propAbstract',
+    'ocat_summary': 'ocatList',
+    'ocat_details': 'ocatDetails',
+    'archive_file_list': 'archiveFileList'}
+
+# Units copied from https://github.com/jzuhone/pycda/blob/
+# 5a4261328eab989bab91bed17f426ad17d876988/pycda/obscat.py#L38
+OCAT_UNITS = {
+    "app_exp": "ks",
+    "count_rate": "s**-1",
+    "est_cnt_rate": "s**-1",
+    "evfil_lo": "keV",
+    "evfil_ra": "keV",
+    "exp_time": "ks",
+    "f_time": "s",
+    "forder_cnt_rate": "s**-1",
+    "soe_roll": "degree",
+    "x_sim": "mm",
+    "y_off": "arcmin",
+    "z_off": "arcmin",
+    "z_sim": "mm",
+}
+
+
+RETURN_TYPE_DOCS = """If ``return_type='auto'`` the return type is determined by the rules:
+
+    - If ``obsid`` is provided
+      AND the obsid corresponds to an integer
+      AND the returned result has a single row
+      THEN the return type is ``dict``
+      ELSE the return tuple is a ``Table``.
+
+    If ``return_type='table'`` then always return a ``Table``."""
+
+CDA_PARAM_DOCS = """Additional function args for CDA search parameters::
+
+        instrument=ACIS,ACIS-I,ACIS-S,HRC,HRC-I,HRC-S
```

Comment:
I'm not sure where the most useful place is to make it clear that the local search and the non-local search are against different column names?